### DISCUSSION TITLE
how to realize multi tenancy using keycloak

### DISCUSSION THREAD

**Discussion by amadereginald** [keycloak/keycloak#12518] [Q&A]
Hi all, 

I want to use keycloak to secure our platform using SSO. 
There are users from multiple organizations 

1. in pilot phase using 1 instance of 1 application 
2. in future using 1 instance of many applications

There is a ongoing discussion to approach this use case 

1. the first proposal is to use keycloak's  group concept and 1 realm
2. the second proposal is to use keycloak's realm concept and create as many realms as participating organizations

What is the best practice?

**Comment by xgp** [comment]
Another approach altogether would be to use this extension that adds an Organization model to Keycloak: 
https://github.com/p2-inc/keycloak-orgs
https://keycloak.discourse.group/t/keycloak-multi-tenancy-extensions-for-saas-applications/15426
